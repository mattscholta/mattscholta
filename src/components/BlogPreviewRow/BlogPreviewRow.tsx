import * as React from 'react';
import { color } from '@barguide/style-guide';
import classnames from 'classnames';
import Link from 'next/link';

import { Star } from '@src/icons';
import { analytics } from '@src/utils/analytics';

import styles from './styles.module.css';

export interface BlogPreviewRowProps {
  className?: string;
  post: any;
  small: boolean;
}

/**
 * @name BlogPreviewRow
 * @description Stateless Component / Pure Function
 */
const BlogPreviewRow: React.FC<BlogPreviewRowProps> = (props) => {
  const { className, post } = props;
  const { slug, sticky, title } = post;

  // Setup
  const link = `/posts/${slug}`;

  // Styles
  const css = classnames('u-flex', styles.component, className);
  const cssCol = classnames('u-flex u-flex-column u-flex-grow', styles.details);

  // Handlers
  const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    const { currentTarget } = event;
    const { gaLabel } = currentTarget.dataset;

    analytics.event('blog-preview-row', 'click', gaLabel);
  };

  // Markup
  const renderImage = () => {
    if (!post.images[0].url) return null;

    return (
      <div className={styles.image}>
        <img
          alt={title || ''}
          height={1250}
          loading="lazy"
          src={post.images[0].url}
          title={`${title}, Read more` || ''}
          width={1880}
        />
      </div>
    );
  };

  return (
    <Link href={link}>
      <a className={css} data-ga-label={slug} onClick={onClick}>
        {renderImage()}
        <div className={cssCol}>
          <h2>
            {sticky && <Star fill={color.gray[7]} height={16} width={16} />}
            {title}
          </h2>
          <span className={styles.link}>Read More &#xbb;</span>
        </div>
      </a>
    </Link>
  );
};

export { BlogPreviewRow };
