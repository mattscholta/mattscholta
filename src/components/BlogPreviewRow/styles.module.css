.component {
  color: var(--spot-grey-8);
  flex: 0 0 400px;
  margin: 0 0 var(--grid-4x);
  min-width: 320px;
}

.component:hover {
  text-decoration: none;
}

.link {
  color: var(--spot-primary);
  display: block;
  font-size: 0.8em;
  transition: color 200ms ease-out;
}

.component:hover .link {
  color: var(--spot-primary);
}

.component h2 {
  color: var(--spot-grey-7);
  font-size: 1em;
  margin-bottom: var(--grid-1xs);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.component svg {
  margin-right: var(--grid-1x);
}

.details {
  justify-content: center;
  overflow: hidden;
}

.image {
  height: auto;
  margin-right: var(--grid-2x);
  max-width: 100px;
  text-transform: uppercase;
}
