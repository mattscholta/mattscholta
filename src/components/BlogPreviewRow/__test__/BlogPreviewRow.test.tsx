import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { render, RenderResult } from '@testing-library/react';

import { BlogPreviewRow, BlogPreviewRowProps } from '../BlogPreviewRow';

describe('BlogPreviewRow', () => {
  let props: BlogPreviewRowProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <MemoryRouter>
      <BlogPreviewRow {...props} />
    </MemoryRouter>
  );

  beforeEach(() => {
    props = {
      className: '',
      post: {
        __typename: 'BlogPost',
        date: '',
        excerpt: '',
        URL: '',
        content: '',
        featured_image: '',
        modified: '',
        post_thumbnail: {
          URL: '',
          __typename: 'BlogPostThumb',
          height: 100,
          width: 100
        },
        slug: '',
        sticky: false,
        title: ''
      },
      small: false
    };

    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
