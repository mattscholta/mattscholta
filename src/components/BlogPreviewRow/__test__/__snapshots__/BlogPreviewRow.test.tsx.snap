// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPreviewRow matches snapshot 1`] = `
NodeList [
  <a
    class="u-flex component"
    data-ga-label=""
    href="/posts/"
  >
    <div
      class="u-flex u-flex-column u-flex-grow details"
    >
      <h2>
        
      </h2>
      <span
        class="link"
      >
        Read More Â»
      </span>
    </div>
  </a>,
]
`;
